### PR TITLE
Kevin

### PR DESCRIPTION
**Pull request by Kevins35-de**
implemented authenticationMiddleware and authentication routes

### CODE REVIEW

File: API/Routes/authenticationRoute.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
+import {Request, Response, Router} from "express";
+import bcrypt from 'bcryptjs';
+import crypto from "crypto";
+import {authenticate} from "../authenticationMiddleware.js";
+import {User} from "../../Database/Mapper/Entities/user.js";
+import {Session} from "../../Database/Mapper/Entities/session.js";
+import {Permission_Techcode} from "../../Database/Mapper/Techcodes/Permission_Techcode";
```
Comment (SSJGodGogeta) [inline API/Routes/authenticationRoute.ts:7]: .js

Comment (Kevins35-de) [reply API/Routes/authenticationRoute.ts]: that would have been an anoying bug to find...

File: API/Routes/authenticationRoute.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+        await user.save();
+
+        // generate the session and store it in the database
+        const session: Session = await createNewSession(req.ip, user!)
+
+        // Set the token as a secure HttpOnly cookie
+        setSessionCookie(res, session.token);
+
+        res.status(200).json({ message: "registering successful" });
```
Comment (SSJGodGogeta) [inline API/Routes/authenticationRoute.ts:79]: use sendResponseAsJson instead of using res.status......

File: API/authenticationMiddleware.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import {NextFunction, Request, Response} from "express";
+import {Session} from "../Database/Mapper/Entities/session";
```
Comment (SSJGodGogeta) [inline API/authenticationMiddleware.ts:2]: Add .js to session

**Review by SSJGodGogeta — APPROVED** [review]

**Pull request merged**